dbms/src/Storages/Transaction/ProxyFFI.cpp

```diff
@@ -157,6 +158,188 @@ uint8_t TryFlushData(EngineStoreServerWrap * server, uint64_t region_id, uint8_t
     }
 }
 
+RawCppPtr CreateWriteBatch(const EngineStoreServerWrap * dummy)
+{
+    UNUSED(dummy);
+    return GenRawCppPtr(new UniversalWriteBatch(), RawCppPtrTypeImpl::WriteBatch);
+}
+
+void WriteBatchPutPage(RawVoidPtr ptr, BaseBuffView page_id, BaseBuffView value)
+{
+    LOG_TRACE(&Poco::Logger::get("ProxyFFI"), fmt::format("FFI write page {}", UniversalPageId(page_id.data, page_id.len)));
+    auto * wb = reinterpret_cast<UniversalWriteBatch *>(ptr);
+    MemoryWriteBuffer buf(0, value.len);
+    buf.write(value.data, value.len);
+    auto data_size = buf.count();
+    assert(data_size == value.len);
+    wb->putPage(UniversalPageId(page_id.data, page_id.len), 0, buf.tryGetReadBuffer(), data_size);
```

Comment:
Is the memory of `page_id` `value` generated by `GenCppRawString`?

Reply:
It is just a view of a proxy memory.

Reply:
so when the proxy uses uni ps WriteBatch, it needs to make sure these memory are available until the writebatch is actually written to uni ps?

Reply:
I think we can add some comments about this

Reply:
`value` is a view of proxy memory. But its' data is copied to `MemoryWriteBuffer`, and it will be managed by `WriteBatch`.

Reply:
I get it